src/tls/openssl/ocsp.c

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file openssl/ocsp.c Online Certificate Status Protocol API
+ *
+ * Copyright (C) 2025 Commend.com - m.fridrich@commend.com
+ */
+#if defined(WIN32)
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+#endif
+#include <openssl/ocsp.h>
+#include <openssl/ssl.h>
+#include <re_types.h>
+#include <re_fmt.h>
+#include <re_sa.h>
+#include <re_srtp.h>
+#include <re_tls.h>
+#include "tls.h"
+
+#define DEBUG_MODULE "tls/ocsp"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+static OCSP_RESPONSE *get_ocsp_response(SSL *ssl, long *resp_len)
+{
+	int rsp_status;
+	OCSP_RESPONSE *rsp;
+	const unsigned char *ocsp_resp;
+	*resp_len = SSL_get_tlsext_status_ocsp_resp(ssl, &ocsp_resp);
+
+	if (*resp_len <= 0 || !ocsp_resp)
+		return NULL;
+
+	rsp = d2i_OCSP_RESPONSE(NULL, &ocsp_resp, *resp_len);
+	if (!rsp)
+		return NULL;
+
+	rsp_status = OCSP_response_status(rsp);
+	if (rsp_status != OCSP_RESPONSE_STATUS_SUCCESSFUL) {
+		OCSP_RESPONSE_free(rsp);
+		return NULL;
+	}
+
+	return rsp;
+}
+
+
+static int get_certificates(SSL *ssl, X509 **cert, X509 **issuer)
+{
+	X509_NAME *iname;
+	STACK_OF(X509) *peer_chain;
+
+#if OPENSSL_VERSION_MAJOR >= 3
+	*cert = SSL_get1_peer_certificate(ssl);
+#else
+	*cert = SSL_get_peer_certificate(ssl);
+#endif
```

Comment:
could this be a good time to drop support for OpenSSL 1.1.1 ?

Reply:
I'm pretty sure we still have systems using OpenSSL 1.1.1.

If you want to discuss this, I think opening a GitHub discussion in baresip or re and tagging all major contributors and all maintainers could be good.